src/graph/validator/MatchValidator.cpp

```diff
@@ -185,7 +188,7 @@ Status MatchValidator::buildPathExpr(const MatchPath *path,
   pathBuild->add(InputPropertyExpression::make(pool, nodeInfos.back().alias));
   pathInfo.pathBuild = std::move(pathBuild);
   pathInfo.anonymous = false;
-  pathInfo.alias = *pathAlias;
+  pathInfo.alias = pathAlias == nullptr ? path->toString() : *pathAlias;
   return Status::OK();
```

Comment:
These codes are confusing.
When the path has no alias, why does the `PathBuild` expression need to be built, and why is the anonymous flag not set to true?

Reply:
match (a:player) return (a)-[]->() . the pattern in return has no alias, but needs to output path。the variable anonymous is useless

Reply:
If an anonymous pattern is set to a non-anonymous pattern, or vice versa, how do you ensure that other code using that data structure doesn't misuse it?

Reply:
fixed